### ISSUE TITLE
[QUESTION] Does the configuration of pino-http also configure pino?

### ISSUE THREAD

**Issue by dlecan** [iamolegga/nestjs-pino#424]
Hello,

I'm using the syntax shown in https://github.com/iamolegga/nestjs-pino#asynchronous-configuration

```javascript
@Module({
  imports: [
    LoggerModule.forRootAsync({
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: async (config: ConfigService) => {
        return {
          pinoHttp: {
            level: config.level,
            prettyPrint: true
          },
        };
      }
    })
  ],
  ...
})
class TestModule {}
```

I get this kind of logs:

```
{"level":40,"time":1612819694266,"pid":37211,"hostname":"...","context":"..Service","msg":"a message"}
[1612819713135] INFO  (my-app/37211 on truc): request completed
    res: {
      "statusCode": 201,
      "headers": {
        "x-powered-by": "Express",
        "access-control-allow-origin": "*",
        "content-type": "application/vnd.api+json; charset=utf-8",
        "content-length": "91",
        "etag": "W/\"5b-otKW0l0jfPtZ563iVaysTjm6m5s\""
      }
    }
    responseTime: 12
  ...
```

Some of them are pretty-printed, some others aren't.

I also tried to force pino configuration:

```javascript
...
      useFactory: async (config: ConfigService) => {
        return {
          pinoHttp: {
            logger: pino({
              level: config.level,
              prettyPrint: true
            },
            level: config.level,
            prettyPrint: true
          },
      }
    })
  ],
  ...
})
class TestModule {}
```

But no change.

How to get the same configuration for all logs?

- pino@6.11.0
- pino-http@5.5.0
- nestjs-pino@1.3.0

Thank you

**Comment by iamolegga** [comment]
Hi, according to [pino-http docs](https://github.com/pinojs/pino-http#pinohttpopts-stream) yes, it should configure pino as well. If you think this is a bug, please feel free to add some minimal represented example repo, but I would suggest to check pino-http first, because this is more related to that lib.

**Comment by dlecan** [comment]
I finally had to debug `PinoLogger` to understand what was happening.
I had somewhere else in my application a "static" initialisation `private readonly logger = new PinoLogger(...)` in a class.

This one was being called at the very startup of my application by `import ...` mechanism, much earlier than my Pino configuration.

Pino was thus being initialised with default configuration as `outOfContext` [in this class](https://github.com/iamolegga/nestjs-pino/blob/master/src/PinoLogger.ts#L17) :disappointed: 

I removed this manual instantiation of the `PinoLogger` and it's working perfectly!

So, the answer of my question is: don't do `new PinoLogger(...)` :wink:
